src/backend/impl/src/repositories/user_profile_repository.rs

```diff
@@ -120,6 +129,8 @@ impl UserProfileRepository for UserProfileRepositoryImpl {
 
         STATE.with_borrow_mut(|s| {
             s.profiles.insert(user_id, user_profile.clone());
+            s.username_index
+                .insert(user_profile.username.clone(), user_id);
```

Comment:
where do we remove the old username from the index?

Reply:
That's a good question. I talked with Wenzel about this a while back and he liked the idea of old usernames continuing to work.

We don't actually have any functionality yet that needs this yet, but maybe in the future we'll be able to look up reviews by reviewer username and there it could be nice if all old usernames would still work. Since we have the username history we can probably do this easily enough, but if we do that we should make sure that old usernames are "reserved" or it will get confusing.

What do you think about this?

Reply:
If we decide to keep the old usernames in the index, we should at least state that "it will still be possible to identify you and your reviews by your old username(s)" in the UI form just for clarity.
This approach is also simple, since the service already has the username check in the update profile and prevents the users from choosing an old username.

From an end user perspective, it may probably sound weird to be able to retrieve the reviews by a username that currently doesn't belong to anyone, but we can probably solve it with some UI messages. Let's keep go for it

Reply:
> we should at least state that "it will still be possible to identify you and your reviews by your old username(s)" in the UI form just for clarity.

Agreed, we should make sure it's clear to users, and possibly also display their old usernames.

> From an end user perspective, it may probably sound weird to be able to retrieve the reviews by a username that currently doesn't belong to anyone.

That's true. Maybe we should spark this conversation up again with Wenzel and discuss it further.